File: ecommerce-project/02-backend/spring-boot-ecommerce/src/main/java/com/kmorath/ecommerce/dto/Purchase.java
package com.kmorath.ecommerce.dto;

import java.util.Set;

import com.kmorath.ecommerce.dao.Address;
import com.kmorath.ecommerce.dao.Customer;
import com.kmorath.ecommerce.dao.Order;
import com.kmorath.ecommerce.dao.OrderItem;

import lombok.Data;

@Data
public class Purchase {

    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;

}

File: ecommerce-project/03-frontend/angular-ecommerce/src/app/DTO/purchase.ts
import { OrderItem } from './order-item';
import { Order } from './order';
import { Address } from './address';
import { Customer } from './customer';

export class Purchase {
  customer: Customer;
  billingAddress: Address;
  shippingAddress: Address;
  order: Order;
  orderItems: OrderItem[];
}
